packages/database/data-migrations/20240610055828_adds_app_and_website_status_indicators/data-migration.ts

```diff
@@ -0,0 +1,88 @@
+import { PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+const main = async () => {
+  await prisma.$transaction(
+    async (tx) => {
+      const startTime = Date.now();
+      console.log("Starting data migration...");
+
+      // Retrieve all environments with widget setup completed
+      const environmentsWithWidgetSetupCompleted = await tx.environment.findMany({
+        where: {
+          widgetSetupCompleted: true,
+        },
+        select: {
+          id: true,
+        },
+      });
+
+      console.log(
+        `Found ${environmentsWithWidgetSetupCompleted.length} environments with widget setup completed.`
+      );
+
+      if (environmentsWithWidgetSetupCompleted.length > 0) {
+        const environmentIds = environmentsWithWidgetSetupCompleted.map((env) => env.id);
+
+        // Fetch survey counts in a single query
+        const surveyCounts = await tx.survey.groupBy({
+          by: ["environmentId", "type"],
+          where: {
```

Comment:
I think to be fully correct we should also check if these surveys also got at least one display. Otherwise we don't know if the app/website was connected.